File: 231121/1이 3개 이상 있는 위치/place-more-than-3-ones.py
N=int(input())

arr=[list(map(int,input().split())) for _ in range(N)]

dxs,dys=[0,1,0,-1],[1,0,-1,0]

def in_range(x,y):
    return 0<=x and x<N and 0<=y and y<N

ans=0

for x in range(N):
    for y in range(N):
        cnt=0
        for dx,dy in zip(dxs,dys):
            nx,ny=x+dx,y+dy
            if in_range(nx,ny) and arr[nx][ny]==1:
                cnt+=1
        if cnt>=3:
            ans+=1

print(ans)

File: 251210/1이 3개 이상 있는 위치/place-more-than-3-ones.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const grid = input.slice(1).map((cell)=>cell.split(' ').map(Number));

const dx = [1,0,-1,0], dy = [0,-1,0,1];

function isValid(x,y){
    return 0<=x && x<n && 0<=y && y<n;
}

let ans =0;

for(let x=0; x<n; x++){
    for(let y=0; y<n; y++){
        let cnt =0;
        for(let dir=0; dir<=3; dir++){
            let nextX = x+dx[dir], nextY = y+dy[dir];
            if(isValid(nextX,nextY) && grid[nextX][nextY]===1){
                cnt++;
            }
        }
        if(cnt>=3){
            ans++;
        }
    }
}

console.log(ans)